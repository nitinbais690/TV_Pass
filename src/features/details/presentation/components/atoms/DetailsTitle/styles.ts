import { StyleSheet } from 'react-native';
import { appFonts } from 'core/styles/AppStyles';

export const styles = () => {
    return StyleSheet.create({
        titleStyle: {
            color: '#ECECEC',
            fontSize: appFonts.lg,
            fontFamily: appFonts.bold,
            fontWeight: '900',
            textTransform: 'uppercase',
        },
    });
};
