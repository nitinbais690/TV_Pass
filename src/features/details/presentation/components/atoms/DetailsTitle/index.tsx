import { appFontStyle } from 'core/styles/AppStyles';
import React from 'react';
import { Text } from 'react-native';
import { styles } from './styles';

export default function DetailsTitle(props: { name: string | undefined; style?: {} }) {
    let titleStyles = styles();
    let fontStyle = appFontStyle;
    return (
        <Text style={[fontStyle.header2, titleStyles.titleStyle, props.style]} numberOfLines={2} ellipsizeMode={'tail'}>
            {props.name ? props.name : ' '}
        </Text>
    );
}
