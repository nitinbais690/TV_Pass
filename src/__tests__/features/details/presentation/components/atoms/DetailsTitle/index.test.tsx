import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import { StyleSheet } from 'react-native';
import DetailsTitle from 'features/details/presentation/components/atoms/DetailsTitle';

jest.mock('core/presentation/hooks/use-app-colors');

describe('detailsTitle', () => {
    const title: string = 'Inception';

    test('render with default style', () => {
        const { container, queryByText } = render(<DetailsTitle name={title} />);
        expect(queryByText(title)).toBeDefined();
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render with custom style', () => {
        const styles = StyleSheet.create({
            lowerCase: {
                textTransform: 'lowercase',
            },
        });

        const { container, queryByText } = render(<DetailsTitle name={title} style={styles.lowerCase} />);
        expect(queryByText(title)).toBeDefined();
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render with empty text', () => {
        const { container } = render(<DetailsTitle name={''} />);
        expect(container).toBeDefined();
        expect(toJSON(container)).toMatchSnapshot();
    });
});
