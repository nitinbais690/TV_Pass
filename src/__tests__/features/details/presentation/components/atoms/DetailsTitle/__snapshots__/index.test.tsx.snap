// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detailsTitle render with custom style 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={2}
    style={
      Array [
        Object {
          "fontFamily": "ProximaNova-Black",
          "fontSize": 40,
          "fontWeight": "900",
        },
        Object {
          "color": "#ECECEC",
          "fontFamily": "ProximaNova-Black",
          "fontSize": 28,
          "fontWeight": "900",
          "textTransform": "uppercase",
        },
        Object {
          "textTransform": "lowercase",
        },
      ]
    }
  >
    Inception
  </Text>
</View>
`;

exports[`detailsTitle render with default style 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={2}
    style={
      Array [
        Object {
          "fontFamily": "ProximaNova-Black",
          "fontSize": 40,
          "fontWeight": "900",
        },
        Object {
          "color": "#ECECEC",
          "fontFamily": "ProximaNova-Black",
          "fontSize": 28,
          "fontWeight": "900",
          "textTransform": "uppercase",
        },
        undefined,
      ]
    }
  >
    Inception
  </Text>
</View>
`;

exports[`detailsTitle render with empty text 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={2}
    style={
      Array [
        Object {
          "fontFamily": "ProximaNova-Black",
          "fontSize": 40,
          "fontWeight": "900",
        },
        Object {
          "color": "#ECECEC",
          "fontFamily": "ProximaNova-Black",
          "fontSize": 28,
          "fontWeight": "900",
          "textTransform": "uppercase",
        },
        undefined,
      ]
    }
  >
     
  </Text>
</View>
`;
